### ISSUE TITLE
ONNX export failed with Conv1d with compiled TorchScript model

### ISSUE THREAD

**Issue by Oktai15** [pytorch/pytorch#35293]
## 🐛 Bug

ONNX export failed with Conv1d with compiled TorchScript model

## To Reproduce

I have toy model:

```python
class OnnxDecoder(nn.Module):
    def __init__(self, *, feat_in, num_classes, init_mode="xavier_uniform"):
        super().__init__()

        self._feat_in = feat_in
        self._num_classes = num_classes

        self.decoder_layers = nn.Sequential(
            nn.Conv1d(self._feat_in, self._num_classes, kernel_size=1, bias=True)
        )
        self.apply(lambda x: init_weights(x, mode=init_mode))

    def forward(self, encoder_output):
        o = self.decoder_layers(encoder_output).transpose(1, 2)
        return functional.log_softmax(o, dim=-1)


class OnnxModel(nn.Module):
    def __init__(self, vocab_size):
        super().__init__()

        self.decoder = OnnxDecoder(
            feat_in=1024,
            num_classes=vocab_size,
        )

    def forward(self, x: torch.Tensor):
        x = torch.flatten(x, start_dim=1)[:, :1024]
        x = x.view(-1, 1024, 1)
        return self.decoder(x)
```

I try to convert it to ONNX using TorchScript (because my real model (not toy) have if-cond, for-loop and conv1d as well):

```python
model = OnnxModel(len(vocabulary)).cpu()
model.eval()

model = torch.jit.script(model)

batch_size = 8
dim_size = 64
max_len = 2000

inp = torch.randn(batch_size, dim_size, max_len)

outs = model(inp)

 # Export the model
 torch.onnx.export(
        model=model,
        args=inp,
        f="model.onnx",
        export_params=True,
        opset_version=10,
        do_constant_folding=True,
        input_names=['inps'],
        output_names=['output'],
        example_outputs=outs,
        dynamic_axes={
            'inps': {0: 'batch_size', 2: 'max_len'},
            'output': {0: 'batch_size', 1: 'out_size'}
        }
 )
```

I got following error:
```
/root/.../site-packages/torch/onnx/utils.py:655: UserWarning: ONNX export failed on ATen operator conv1d because torch.onnx.symbolic_opset10.conv1d does not exist
  .format(op_name, opset_version, op_name))
Traceback (most recent call last):
  File "onnx_export.py", line 50, in <module>
    main()
  File "onnx_export.py", line 44, in main
    'output': {0: 'batch_size', 1: 'out_size'}
  File "/root/.../site-packages/torch/onnx/__init__.py", line 148, in export
    strip_doc_string, dynamic_axes, keep_initializers_as_inputs)
  File "/root/.../site-packages/torch/onnx/utils.py", line 66, in export
    dynamic_axes=dynamic_axes, keep_initializers_as_inputs=keep_initializers_as_inputs)
  File "/root/.../site-packages/torch/onnx/utils.py", line 416, in _export
    fixed_batch_size=fixed_batch_size)
  File "/root/.../site-packages/torch/onnx/utils.py", line 296, in _model_to_graph
    fixed_batch_size=fixed_batch_size, params_dict=params_dict)
  File "/root/.../site-packages/torch/onnx/utils.py", line 135, in _optimize_graph
    graph = torch._C._jit_pass_onnx(graph, operator_export_type)
  File "/root/.../site-packages/torch/onnx/__init__.py", line 179, in _run_symbolic_function
    return utils._run_symbolic_function(*args, **kwargs)
  File "/root/.../site-packages/torch/onnx/utils.py", line 656, in _run_symbolic_function
    op_fn = sym_registry.get_registered_op(op_name, '', opset_version)
  File "/root/.../site-packages/torch/onnx/symbolic_registry.py", line 91, in get_registered_op
    return _registry[(domain, version)][opname]
KeyError: 'conv1d'
```

## Environment

PyTorch version: 1.4.0
Is debug build: No
CUDA used to build PyTorch: 10.1

OS: Debian GNU/Linux 
CMake version: version 3.7.2

Python version: 3.7
Is CUDA available: Yes

Nvidia driver version: 430.14

Versions of relevant libraries:
[pip] numpy==1.18.1
[pip] torch==1.4.0
[pip] torchvision==0.5.0
[conda] blas                      1.0                         mkl
[conda] mkl                       2020.0                      166
[conda] mkl-service               2.3.0            py37he904b0f_0
[conda] mkl_fft                   1.0.15           py37ha843d7b_0
[conda] mkl_random                1.1.0            py37hd6b4f25_0
[conda] torch                     1.4.0                    pypi_0    pypi
[conda] torchvision               0.5.0                    pypi_0    pypi

cc @suo @houseroad @spandantiwari @lara-hdr @BowenBao @neginraoof

**Comment by suo** [comment]
cc @spandantiwari @lara-hdr

**Comment by Oktai15** [comment]
@spandantiwari @lara-hdr @suo 

I resolved a problem using very strange approach, because I use old way for TorchScript via `torch.jit.ScriptModule` explicitly. I mean I changed only one line in my example:

```python
class OnnxDecoder(torch.jit.ScriptModule): # modification is here
    def __init__(self, *, feat_in, num_classes, init_mode="xavier_uniform"):
        super().__init__()

        self._feat_in = feat_in
        self._num_classes = num_classes

        self.decoder_layers = nn.Sequential(
            nn.Conv1d(self._feat_in, self._num_classes, kernel_size=1, bias=True)
        )
        self.apply(lambda x: init_weights(x, mode=init_mode))

    def forward(self, encoder_output):
        o = self.decoder_layers(encoder_output).transpose(1, 2)
        return functional.log_softmax(o, dim=-1)


class OnnxModel(nn.Module):
    def __init__(self, vocab_size):
        super().__init__()

        self.decoder = OnnxDecoder(
            feat_in=1024,
            num_classes=vocab_size,
        )

    def forward(self, x: torch.Tensor):
        x = torch.flatten(x, start_dim=1)[:, :1024]
        x = x.view(-1, 1024, 1)
        return self.decoder(x)

```

At the first, I tried to do subclass `OnnxModel` from `torch.jit.ScriptModule`, but I got error. It means I need to convert my inner class to `torch.jit.ScriptModule`, but general class should be `nn.Module` -- ambiguous behavior. 

Could you clarify how to work with ONNX and TorchScript models correctly?

**Comment by garymm** [comment]
Seems fixed in PyTorch 1.10.0 (or earlier).
